TradeBeacons/data/scripts/systems/tradingoverview.lua

```diff
@@ -102,28 +102,64 @@ function getTradeBeaconScanRange(seed, rarity)
 end
 
 function requestSectorsData(caller)
-    if not caller then
+    local player
```

Comment:
this is where things get a little tricky:

it turns out that all the galaxy-wide trading data is kept on Player objects. When a client initiates a sector data update, that's no big deal, we know which Player object to get trading data from (callingPlayer). But for server-side code we don't have a calling Player. So we need to figure one out. If it's a player controlled ship, that's relatively easy. But for alliance controlled ships, that's trickier.